instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java

```diff
@@ -189,4 +205,164 @@ public void onResponse(Call call, Response response) {
               span -> span.hasName("child").hasKind(SpanKind.INTERNAL).hasParent(trace.getSpan(0)));
         });
   }
+
+  // TODO: replace the resend tests in the AbstractHttpClient with these
+
+  @Test
+  void new_basicRequestWith1Redirect() throws Exception {
+    URI uri = resolveAddress("/redirect");
+
+    int responseCode = sendRequest(buildRequest("GET", uri, emptyMap()), "GET", uri, emptyMap());
+
+    assertThat(responseCode).isEqualTo(200);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttribute(SemanticAttributes.HTTP_URL, uri.toString()),
+                span ->
+                    span.hasName("test-http-server")
+                        .hasKind(SpanKind.SERVER)
+                        .hasParent(trace.getSpan(0))),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttribute(
+                            SemanticAttributes.HTTP_URL, uri.resolve("/success").toString())
+                        .hasAttribute(SemanticAttributes.HTTP_RESEND_COUNT, 1L),
```

Comment:
👍🏻

Reply:
Is it important to verify that the response code was 302 or 301 or whatever? There are attributes for those as well, right?

Reply:
It is - but that's one of the things I'd like to leave for one of the next PRs, the one about replacing the retry tests in `AbstractHttpClientTest`